src/app/Shared/Services/Api.service.tsx

```diff
@@ -90,15 +90,18 @@ export class ApiService {
     );
   }
 
-  createRecording(
-    { recordingName, events, duration  }: { recordingName: string; events: string; duration?: number }
+  createRecording({ recordingAttributes }: { recordingAttributes: RecordingAttributes }
     ): Observable<boolean> {
       const form = new window.FormData();
-      form.append('recordingName', recordingName);
-      form.append('events', events);
-      if (!!duration && duration > 0) {
-        form.append('duration', String(duration));
+      form.append('recordingName', recordingAttributes.name);
+      form.append('events', recordingAttributes.events);
+      form.append('toDisk', String(recordingAttributes.options.toDisk));
+      form.append('maxAge', String(recordingAttributes.options.maxAge));
+      form.append('maxSize', String(recordingAttributes.options.maxSize));
```

Comment:
For validation purposes, `toDisk/maxAge/maxSize` should perform a check similar to what is used for `duration` below, ensuring that the values are defined before they are included in the form contents. Although the component that provides these values is currently written to first retrieve default values from the backend and then pass them through the attributes configuration object when the data flow reaches here, this function should still validate the incoming data for robustness.